src/transformers/utils/quantizers.py

```diff
@@ -0,0 +1,787 @@
+        model.eval()
+
+    def is_model_serializeable(self, model: PreTrainedModel) -> bool:
+        return False
+
+    def is_model_trainable(self, model: PreTrainedModel) -> bool:
+        return False
+
+    def set_torch_dtype(self, torch_dtype: torch.dtype) -> torch.dtype:
+        return torch_dtype
+
+    def update_device_map(self, device_map: DeviceMap) -> DeviceMap:
+        return device_map
+
+    def get_special_dtypes_update(self, model, torch_dtype: torch.dtype) -> Dict[str, torch.dtype]:
+        """returns dtypes for modules that are not quantized"""
+        return {
+            name: torch_dtype
+            for name, _ in model.named_parameters()
+            if any(m in name for m in self.modules_to_not_convert)
+        }
+
+    def adjust_target_dtype(self, torch_dtype: torch.dtype) -> torch.dtype:
+        return torch_dtype
+
+    def adjust_max_memory(self, max_memory: MaxMemory) -> MaxMemory:
+        """adjust max_memory argument for infer_auto_device_map() if extra memory is needed for quantization"""
+        return max_memory
+
+    def validate_device_map(self, device_map: DeviceMap):
+        """validates device map after process_model_before_weight_loading() and infer_auto_device_map()"""
+        ...
+
+    def update_mismatched_keys(
+        self, unexpected_keys: List[str], missing_keys: List[str]
+    ) -> Tuple[List[str], List[str]]:
+        """removes auxiliary quantization components from unexpected_keys, missing_keys"""
+        # TODO: make this a universal function driven by class param `aux_keys`
+        return unexpected_keys, missing_keys
+
+    def check_quantized_param(self, *args, **kwargs) -> bool:
+        """
+        checks if a loaded state_dict component is part of quantized paramv + some validation; only defined if
+        requires_parameters_quantization == True
+        """
+        return False
+
+    def create_quantized_param(self, *args, **kwargs) -> torch.nn.Parameter:
+        """
+        takes needed components from state_dict and creates quantized param; only applicable if
+        requires_parameters_quantization == True
+        """
+        if not self.requires_parameters_quantization:
+            raise AttributeError(
+                "`.create_quantized_param()` method is not supported by quantizer class {self.__cls__}."
+            )
+
+
+class GPTQHFQuantizer(HFQuantizer):
+
+    """
+    quantization method:
+        before weight loading converts layers into GPTQ layers.
+    saving: from state_dict() as any normal model
+        each quantized weight stores in state dict: .qweight, .qzeros, .scales
+    loading: preprocess model into special GPTQ layers, then load into state_dict as usual
+    """
+
+    require_low_cpu_mem_usage = True
+    method = QuantizationMethod.GPTQ
+
+    def __init__(self, quantization_config: QuantizationConfig, **kwargs):
+        super().__init__(quantization_config, **kwargs)
```

Comment:
shouldn't we call `validate_environment` inside init as well? What do you think?

Reply:
The present choice of arguments to `validate_environment()` looks awkward to me: torch_dtype, from_tf, from_flax. 
It was driven by validation requirements of present quantizers. Passing them via quantization_config_parser.get_quantizer() would be even more awkward. 

I would rather keep the functions separate, but pass the whole context to the validation fn. This would make life easier for the future quantizers.

Reply:
@younesbelkada, How would you like the idea of passing the whole context of `from_pretrained()` to `HFQuantizer` methods? It is not very pythonic, but this way we'd be able to combine `validate_environment(), set_torch_dtype(), update_device_map()` into one method.

Reply:
> I would rather keep the functions separate, but pass the whole context to the validation fn. This would make life easier for the future quantizers.

Yes OK got it, makes sense, let's keep it as it is i would say. I personally really like the fact that everything is separated and makes things very clear, validate dtype, device_map, etc. Let's wait for other maintainers to see their thoughts